[CI] CancellableTasksIT testChildrenTasksCancelledOnTimeout failing

**Build Scans:**
- [elasticsearch-intake #24730 / part1](https://gradle-enterprise.elastic.co/s/we45abo56bywk)
- [elasticsearch-periodic #8888 / openjdk21_checkpart1_java-fips-matrix](https://gradle-enterprise.elastic.co/s/uzhjaumnihkiq)

**Reproduction Line:**
```
./gradlew ":server:internalClusterTest" --tests "org.elasticsearch.action.admin.cluster.node.tasks.CancellableTasksIT.testChildrenTasksCancelledOnTimeout" -Dtests.seed=BA219B66736C7297 -Dtests.locale=nso-Latn-ZA -Dtests.timezone=Etc/GMT-10 -Druntime.java=24
```

**Applicable branches:**
9.0

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.action.admin.cluster.node.tasks.CancellableTasksIT),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!(testChildrenTasksCancelledOnTimeout),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: All incoming requests on node [node_t0] should have finished. Expected 0 bytes for requests in-flight but got 1098 bytes; pending tasks [[{
  "node" : "node_t0",
  "id" : 76,
  "type" : "transport",
  "action" : "internal::test_action",
  "description" : "id=439",
  "start_time" : "2025-06-27T19:36:37.132Z",
  "start_time_in_millis" : 1751052997132,
  "running_time" : "1.1m",
  "running_time_in_nanos" : 66369747135,
  "cancellable" : true,
  "cancelled" : false,
  "parent_task_id" : "lXrNGVLnR5iGzioG3xHluQ:26",
  "headers" : { }
}]]
```

**Issue Reasons:**
- [9.0] 2 failures in test testChildrenTasksCancelledOnTimeout (0.4% fail rate in 473 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

I was able to reproduce an error on my machine by setting the timeout to 1 millisecond:
```
diff --git a/server/src/internalClusterTest/java/org/elasticsearch/action/admin/cluster/node/tasks/CancellableTasksIT.java b/server/src/internalClusterTest/java/org/elasticsearch/action/admin/cluster/node/tasks/CancellableTasksIT.java
index 3f3ee0c5598..1c9024c6dce 100644
--- a/server/src/internalClusterTest/java/org/elasticsearch/action/admin/cluster/node/tasks/CancellableTasksIT.java
+++ b/server/src/internalClusterTest/java/org/elasticsearch/action/admin/cluster/node/tasks/CancellableTasksIT.java
@@ -583,7 +583,7 @@ public class CancellableTasksIT extends ESIntegTestCase {
                         }
                     } else {
                         final TransportRequestOptions transportRequestOptions = subRequest.timeout
-                            ? TransportRequestOptions.timeout(TimeValue.timeValueMillis(400))
+                            ? TransportRequestOptions.timeout(TimeValue.timeValueMillis(1))
                             : TransportRequestOptions.EMPTY;
                         transportService.sendRequest(
                             subRequest.node,
```

```
  2> java.lang.AssertionError: All incoming requests on node [node_t5] should have finished. Expected 0 bytes for requests in-flight but got 667 bytes; pending tasks [[{
      "node" : "node_t5",
      "id" : 48,
      "type" : "transport",
      "action" : "internal::test_action",
      "description" : "id=63",
      "start_time" : "2025-06-04T16:42:44.473Z",
      "start_time_in_millis" : 1749055364473,
      "running_time" : "1m",
      "running_time_in_nanos" : 65613587791,
      "cancellable" : true,
      "cancelled" : false,
      "parent_task_id" : "5N6KZHDGTiiBZSjD2Ot9lA:31",
      "headers" : { }
    },{
      "node" : "node_t5",
      "id" : 46,
      "type" : "transport",
      "action" : "internal::test_action",
      "description" : "id=53",
      "start_time" : "2025-06-04T16:42:44.473Z",
      "start_time_in_millis" : 1749055364473,
      "running_time" : "1m",
      "running_time_in_nanos" : 65614233417,
      "cancellable" : true,
      "cancelled" : false,
      "parent_task_id" : "MbXQaXENSdKRuK1C2AlO2A:38",
      "headers" : { }
    }]]
        at __randomizedtesting.SeedInfo.seed([88E20B7E9E3A1AC2:CCFB6194F0352C92]:0)
        at org.elasticsearch.test.InternalTestCluster.lambda$assertRequestsFinished$41(InternalTestCluster.java:2596)
        at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:1481)
        at org.elasticsearch.test.InternalTestCluster.assertRequestsFinished(InternalTestCluster.java:2587)
        at org.elasticsearch.test.InternalTestCluster.assertAfterTest(InternalTestCluster.java:2559)
        at org.elasticsearch.test.ESIntegTestCase.afterInternal(ESIntegTestCase.java:605)
        at org.elasticsearch.test.ESIntegTestCase.cleanUpCluster(ESIntegTestCase.java:2544)
        at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)
```

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -2,9 +2,6 @@ tests:
 - class: org.elasticsearch.xpack.apmdata.APMYamlTestSuiteIT
   method: test {yaml=/10_apm/Test template reinstallation}
   issue: https://github.com/elastic/elasticsearch/issues/116445
-- class: org.elasticsearch.action.admin.cluster.node.tasks.CancellableTasksIT
-  method: testChildrenTasksCancelledOnTimeout
-  issue: https://github.com/elastic/elasticsearch/issues/123568
 - class: org.elasticsearch.xpack.profiling.action.GetStatusActionIT
   method: testWaitsUntilResourcesAreCreated
   issue: https://github.com/elastic/elasticsearch/issues/129486

--- a/server/src/internalClusterTest/java/org/elasticsearch/action/admin/cluster/node/tasks/CancellableTasksIT.java
+++ b/server/src/internalClusterTest/java/org/elasticsearch/action/admin/cluster/node/tasks/CancellableTasksIT.java
@@ -44,7 +44,6 @@ import org.elasticsearch.tasks.TaskInfo;
 import org.elasticsearch.tasks.TaskManager;
 import org.elasticsearch.test.ESIntegTestCase;
 import org.elasticsearch.threadpool.ThreadPool;
-import org.elasticsearch.transport.ReceiveTimeoutTransportException;
 import org.elasticsearch.transport.SendRequestTransportException;
 import org.elasticsearch.transport.Transport;
 import org.elasticsearch.transport.TransportRequestOptions;
@@ -66,7 +65,6 @@ import java.util.concurrent.TimeUnit;
 import java.util.stream.Collectors;
 
 import static org.hamcrest.Matchers.anyOf;
-import static org.hamcrest.Matchers.containsStringIgnoringCase;
 import static org.hamcrest.Matchers.empty;
 import static org.hamcrest.Matchers.equalTo;
 import static org.hamcrest.Matchers.hasSize;
@@ -205,7 +203,7 @@ public class CancellableTasksIT extends ESIntegTestCase {
         } finally {
             allowEntireRequest(rootRequest);
             cancelFuture.actionGet();
-            waitForRootTask(rootTaskFuture, false);
+            waitForRootTask(rootTaskFuture);
             ensureBansAndCancellationsConsistency();
         }
     }
@@ -226,7 +224,7 @@ public class CancellableTasksIT extends ESIntegTestCase {
         assertFalse(cancelFuture.isDone());
         allowEntireRequest(rootRequest);
         assertThat(cancelFuture.actionGet().getTaskFailures(), empty());
-        waitForRootTask(mainTaskFuture, false);
+        waitForRootTask(mainTaskFuture);
         ListTasksResponse cancelError = clusterAdmin().prepareCancelTasks()
             .setTargetTaskId(taskId)
             .waitForCompletion(randomBoolean())
@@ -256,7 +254,7 @@ public class CancellableTasksIT extends ESIntegTestCase {
             cancelFuture.get();
         }
         allowEntireRequest(rootRequest);
-        waitForRootTask(mainTaskFuture, false);
+        waitForRootTask(mainTaskFuture);
         if (waitForCompletion) {
             cancelFuture.actionGet();
         }
@@ -278,7 +276,7 @@ public class CancellableTasksIT extends ESIntegTestCase {
         TaskCancelledException te = expectThrows(TaskCancelledException.class, future);
         assertThat(te.getMessage(), equalTo("parent task was cancelled [by user request]"));
         allowEntireRequest(rootRequest);
-        waitForRootTask(rootTaskFuture, false);
+        waitForRootTask(rootTaskFuture);
         ensureBansAndCancellationsConsistency();
     }
 
@@ -370,20 +368,6 @@ public class CancellableTasksIT extends ESIntegTestCase {
         }
     }
 
-    public void testChildrenTasksCancelledOnTimeout() throws Exception {
-        Set<DiscoveryNode> nodes = clusterService().state().nodes().stream().collect(Collectors.toSet());
-        final TestRequest rootRequest = generateTestRequest(nodes, 0, between(1, 4), true);
-        ActionFuture<TestResponse> rootTaskFuture = client().execute(TransportTestAction.ACTION, rootRequest);
-        allowEntireRequest(rootRequest);
-        waitForRootTask(rootTaskFuture, true);
-        ensureBansAndCancellationsConsistency();
-
-        // Make sure all descendent requests have completed
-        for (TestRequest subRequest : rootRequest.descendants()) {
-            safeAwait(completedLatches.get(subRequest));
-        }
-    }
-
     static TaskId getRootTaskId(TestRequest request) throws Exception {
         SetOnce<TaskId> taskId = new SetOnce<>();
         assertBusy(() -> {
@@ -401,24 +385,19 @@ public class CancellableTasksIT extends ESIntegTestCase {
         return taskId.get();
     }
 
-    static void waitForRootTask(ActionFuture<TestResponse> rootTask, boolean expectToTimeout) {
+    static void waitForRootTask(ActionFuture<TestResponse> rootTask) {
         try {
             rootTask.actionGet();
         } catch (Exception e) {
-            final Throwable cause = ExceptionsHelper.unwrap(
-                e,
-                expectToTimeout ? ReceiveTimeoutTransportException.class : TaskCancelledException.class
-            );
+            final Throwable cause = ExceptionsHelper.unwrap(e, TaskCancelledException.class);
             assertNotNull(cause);
             assertThat(
                 cause.getMessage(),
-                expectToTimeout
-                    ? containsStringIgnoringCase("timed out after")
-                    : anyOf(
-                        equalTo("parent task was cancelled [by user request]"),
-                        equalTo("task cancelled before starting [by user request]"),
-                        equalTo("task cancelled [by user request]")
-                    )
+                anyOf(
+                    equalTo("parent task was cancelled [by user request]"),
+                    equalTo("task cancelled before starting [by user request]"),
+                    equalTo("task cancelled [by user request]")
+                )
             );
         }
     }

